colibri/src/icons.rs

```diff
@@ -119,6 +138,86 @@ async fn query_token_icon_and_extension(
     None
 }
 
+async fn query_uniswap_v3_position_icon(
+    chain_id: u64,
+    token_id: &str,
+    contract_address: &str,
+    globaldb: &globaldb::GlobalDB,
+) -> Option<(Bytes, &'static str)> {
+    let blockchain = SupportedBlockchain::from_chain_id(chain_id)?;
+    let rpc_endpoints = globaldb.get_rpc_nodes(blockchain).await.ok()?;
+    let contract_address = match Address::parse_checksummed(contract_address, None) {
+        Ok(address) => address,
+        Err(_e) => {
+            return None;
+        }
+    };
+    let token_id: U256 = token_id.parse().unwrap();
+
+    // Try all RPC endpoints until one succeeds
+    for rpc_url in rpc_endpoints {
+        let provider = match rpc_url.parse().map(|url| ProviderBuilder::new().on_http(url)) {
```

Comment:
you are destroying and creating the rpcs all the time. Ideally you would have them created and reused in some structure like we have in python the Web3 node manager. What does alloy in the builder?